java/code/src/com/suse/manager/webui/services/pillar/MinionVirtualizationPillarGenerator.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.services.pillar;
+
+import static com.suse.manager.webui.services.SaltConstants.PILLAR_DATA_FILE_EXT;
+import static com.suse.manager.webui.services.SaltConstants.PILLAR_DATA_FILE_PREFIX;
+
+import com.redhat.rhn.common.conf.Config;
+import com.redhat.rhn.common.conf.ConfigDefaults;
+import com.redhat.rhn.domain.server.MinionServer;
+
+import com.suse.manager.webui.utils.SaltPillar;
+
+import org.apache.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for generating pillar data for the virtual hosts
+ */
+public class MinionVirtualizationPillarGenerator implements MinionPillarGenerator {
```

Comment:
We should probably make a test for this

Reply:
sure, though that's not the most interesting test to have here. Covering the whole thing with a cucumber step would be better, though harder

Reply:
Made that test and a cucumber one!